sdk/daml-lf/language/src/main/scala/com/digitalasset/daml/lf/language/PackageInterface.scala

```diff
@@ -160,17 +170,27 @@ private[lf] class PackageInterface(val signatures: PartialFunction[PackageId, Pa
       tyCon: TypeConName,
       consName: EnumConName,
       context: => Reference,
-  ): Either[LookupError, Int] =
-    lookupDataEnum(tyCon, context).flatMap { dataEnumInfo =>
-      dataEnumInfo.dataEnum.constructorRank.get(consName) match {
-        case Some(rank) => Right(rank)
-        case None =>
-          Left(LookupError.NotFound(Reference.DataEnumConstructor(tyCon, consName), context))
-      }
-    }
+  ): Either[LookupError.NotFound, Either[LookupError, Int]] =
+    lookupDataEnum(tyCon, context).fold(
+      notFound => Right(Left(notFound)),
+      dataEnumInfo =>
+        dataEnumInfo.dataEnum.constructorRank.get(consName) match {
+          case Some(rank) =>
+            Right(Right(rank))
+          case None =>
+            Left(LookupError.NotFound(Reference.DataEnumConstructor(tyCon, consName), context))
+        },
+    )
 
   def lookupEnumConstructor(tyCon: TypeConName, consName: EnumConName): Either[LookupError, Int] =
     lookupEnumConstructor(tyCon, consName, Reference.DataEnumConstructor(tyCon, consName))
+      .fold(notFound => Left(notFound), identity)
+
+  def unsafeLookupEnumConstructor(
```

Comment:
why unsafe ?

Reply:
Let me double check - I'd used this name initially for debug/dev purposes and thought I'd changed it

Reply:
Indeed. Sorry for the noise.